core-plugins/src/main/java/io/cdap/plugin/transform/ProjectionTransform.java

```diff
@@ -112,26 +118,28 @@ public ProjectionTransform(ProjectionTransformConfig projectionTransformConfig)
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     // call init so any invalid config is caught here to fail application creation
-    init(pipelineConfigurer.getStageConfigurer().getInputSchema());
+    init(pipelineConfigurer.getStageConfigurer().getInputSchema(), collector);
     Schema outputSchema = null;
     if (pipelineConfigurer.getStageConfigurer().getInputSchema() != null) {
       //validate the input schema and get the output schema for it
-      outputSchema = getOutputSchema(pipelineConfigurer.getStageConfigurer().getInputSchema());
+      outputSchema = getOutputSchema(pipelineConfigurer.getStageConfigurer().getInputSchema(), collector);
     }
     pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
   }
 
   @Override
   public void initialize(TransformContext context) throws Exception {
     super.initialize(context);
-    init(null);
+    init(null, context.getFailureCollector());
```

Comment:
if there are errors at this point, an exception needs to be thrown. This will just ignore the errors, which will cause a confusing error later in transform.

call context.getFailureCollector().getOrThrowException() after this line.

Reply:
Fixed